### PR TITLE
Assassin Tier 1

### PR DESCRIPTION
**Pull request by Terotrous**
Implements Assassin's 3 tier 1 skills.

### CODE REVIEW

**Comment by exectails** [comment]
I'm going to hold off on reviewing your other modifier-including PRs until master is merged in, to make it easier to review them.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs
Comment (exectails) [inline src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs:20]: It feels kinda iffy to create a new throwaway Skill object every tick, doesn't it? Especially because we know this would quickly get copied to all buffs and skills with similar effects. Do officials actually send `Assassin_Behead_DOT`?

Comment (exectails) [inline src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs:20]: It seems like this skill's max level is 1, so there's presumably no point in setting its level.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (exectails) [inline src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:8]: This summary appears to be copy-pasted from another buff.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (exectails) [inline src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:13]: Our coding conventions dictate that consts, statics, and public fields should be formatted in `PascalCase`.

File: src/ZoneServer/Skills/Handlers/Scout/Assassin/Behead.cs
Comment (exectails) [inline src/ZoneServer/Skills/Handlers/Scout/Assassin/Behead.cs:87]: I believe I saw you guys talk about this skill on Discord, and as such I assume you did some research on it, but just for confirmation: Is this how it's supposed to work, with 3 multi hits split up across two skill hits?

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (exectails) [inline src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:32]: Similarly to the skill no longer saying that it consumes a Hasisas Potion, it seems like this also used to
> continuously consume your HP as a side effect

Is this also no longer a thing? Though I did check the buff description, and there the health loss is mentioned.

File: src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs
Comment (exectails) [inline src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs:79]: Am I missing something, or should this be a 1-hit skill? Neither the description nor the bytool data appears to mention 4 hits.

**Review by exectails — CHANGES_REQUESTED** [review]

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs
Comment (Terotrous) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs]: I added an overload to ZC_Hit_Info that takes a SkillId instead of a Skill object (it only uses the SkillId anyway), which allows this to work without creating a skill object for every tick.

Comment (Terotrous) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs]: This is no longer necessary, though I disagree in general.  Even if the skill max level is 1 in the skillDb, levelling it up still works normally if the skill level is increased to be more than 1 (even things like factors display and work properly), so we should probably still handle skill levels correctly.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (Terotrous) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:8]: You got me.  I fixed it.
(I can't believe I did this after fixing so many of them myself in other PRs).

File: src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs
Comment (Terotrous) [reply src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs:79]: It's in the skill tooltip, I posted a screenshot of it in the chat.  Hopefully this links to it?
https://discord.com/channels/273278107773960192/1128408008469266503/1260631576925175840

File: src/ZoneServer/Skills/Handlers/Scout/Assassin/Behead.cs
Comment (Terotrous) [reply src/ZoneServer/Skills/Handlers/Scout/Assassin/Behead.cs:87]: Confirmed by Kenedos that it does work this way.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (Terotrous) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:32]: Definitely seems like this was removed, I note in the old buff description, it also increased your crit damage by proportion to your missing health, but that whole part of the description is gone.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs
Comment (exectails) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs]: While I fully agree with you that we should generally accommodate levels past the maximums, I don't believe this applies to `Assassin_Behead_DOT`, because it's a dummy skill without any additional effects. It doesn't do anything on its own and has no scaling factors whatsoever. As such, I would argue setting a level may cause confusion, making someone think there actually is more to it, when in reality there isn't and likely never will be.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (exectails) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:32]: Just checked the latest description of the buff once more to be absolute certain, and it does still state that it uses HP.
> Increases basic attack speed and critical damage. Reduces HP over time

Unless that effect is not present on officials, we probably want to add it to the buff, right? That will also make it less weird that you no longer need an item to use it.

File: src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs
Comment (exectails) [reply src/ZoneServer/Skills/Handlers/Scout/Assassin/InstantAcceleration.cs:79]: Well, apparently I'm blind. Nevermind me^^

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs
Comment (kenedos) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Hasisas.cs:32]: Checked on official and seems it does not consume hp overtime or upon cast at all.
Buff tooltip still says it does, however.

File: src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs
Comment (Terotrous) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs]: True, though we're not doing it anymore since we're no longer creating the skill object with the new overload.

Comment (exectails) [reply src/ZoneServer/Buffs/Handlers/Scout/Assassin/Behead_Debuff.cs]: True, but it's something we can keep in mind going forward^^

**Pull request merged**
